pkg/resolvers/grbClusterResolver.go

```diff
@@ -86,6 +109,74 @@ func (g *GRBClusterRuleResolver) VisitRulesFor(user user.Info, namespace string,
 			return
 		}
 	}
+	// if we aren't an exception service account, no need to check add sa-specific permissions
+	if !isExceptionServiceAccount(user) {
+		return
+	}
+	isAdmin, err := g.hasAdminPermissions(user)
+	if err != nil {
+		visitor(nil, nil, err)
+		return
+	}
+	if isAdmin {
+		// exception service accounts are considered to have full permissions for the purposes of the clusterRules
+		if !visitRules(nil, adminRules, nil, visitor) {
+			return
+		}
+	}
+}
+
+// hasAdminPermissions checks if a given user is an admin
+func (g *GRBClusterRuleResolver) hasAdminPermissions(user user.Info) (bool, error) {
+	extras := map[string]v1.ExtraValue{}
+	for extraKey, extraValue := range user.GetExtra() {
+		extras[extraKey] = v1.ExtraValue(extraValue)
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+	resourceResponse, err := g.sar.Create(ctx, &v1.SubjectAccessReview{
```

Comment:
Should we add a `nil` check here since the SAR is nil in some cases (tests mostly)?